inc/saiswitch.h

```diff
@@ -788,6 +788,30 @@ typedef enum _sai_switch_attr_t
      */
     SAI_SWITCH_ATTR_AVAILABLE_IPMC_ENTRY,
 
+    /**
+     * @brief Available SNAT entries
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_SNAT_ENTRY,
+
+    /**
+     * @brief Available DNAT entries
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_DNAT_ENTRY,
+
+    /**
+     * @brief Available Double NAT entries
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_SWITCH_ATTR_AVAILABLE_DOBULE_NAT_ENTRY,
```

Comment:
SAI already has a general sai_object_type_get_availability function that allows you to achieve the same CRM functionality

Reply:
Review it with Gouhan. My preference is to do it in separate clean up PR given the scope of changes and its impact on schedule

Reply:
@lguohan , Hi Guohan, please comment here?  Thanks.